src/test/scala/org/apache/spark/sql/delta/OptimisticTransactionSuite.scala

```diff
@@ -93,6 +94,204 @@ class OptimisticTransactionSuite extends QueryTest with SharedSparkSession {
     }
   }
 
+  val A_P1 = "part=p1/a"
+  val B_P1 = "part=p1/b"
+  val C_P2 = "part=p2/c"
+  val D_P2 = "part=p2/d"
+  val E_P3 = "part=p3/e"
+  val F_P3 = "part=p3/f"
+  private val addA_P1 = AddFile(A_P1, Map("part" -> "p1"), 1, 1, dataChange = true)
+  private val addB_P1 = AddFile(B_P1, Map("part" -> "p1"), 1, 1, dataChange = true)
+  private val addC_P2 = AddFile(C_P2, Map("part" -> "p2"), 1, 1, dataChange = true)
+  private val addD_P2 = AddFile(D_P2, Map("part" -> "p2"), 1, 1, dataChange = true)
+  private val addE_P3 = AddFile(E_P3, Map("part" -> "p3"), 1, 1, dataChange = true)
+  private val addF_P3 = AddFile(F_P3, Map("part" -> "p3"), 1, 1, dataChange = true)
+
+  test("block concurrent commit when table is not configured") {
+    withLog(addA_P1 :: addD_P2 :: addD_P2.remove :: Nil) { log =>
+      val tx1 = log.startTransaction()
+      // TX1 reads A_P1
+      tx1.filterFiles()
+
+      val tx2 = log.startTransaction()
+      tx2.filterFiles()
+      // TX2 modifies P2
+      tx2.commit(addC_P2 :: addD_P2.remove :: Nil, Truncate())
+
+      intercept[ConcurrentWriteException] {
+        // this is safe however table was not configured for concurrent writes
+        tx1.commit(addE_P3 :: addF_P3 :: Nil, Truncate())
+      }
+    }
+  }
+
+  test("allow concurrent commit independent partitions") {
+    withConcurrentLog(addA_P1 :: addD_P2 :: addD_P2.remove :: Nil) { log =>
+      val tx1 = log.startTransaction()
+      // TX1 reads A_P1
+      tx1.filterFiles()
```

Comment:
This test makes me a little scared of this change. If tx1 is calling filterFiles, that means the intent of that transaction is to read everything in the table, and so it should logically conflict with tx2.

Reply:
Tx1 reads everything from the table, but the table contains only one partition - **P1** (since D_P2 was added and then removed in init). The TX2 is modifying only **P2**. So Tx1 (reading P1 & modifying P3) have no logical conflict with TX2 (modifying P2). I think this is safe